src/TrifanHarness.cpp

```diff
@@ -76,9 +76,21 @@ int main()
         }
         else if(flight_command == "land")
         {
-            // need handling for what if in transitional or forward flight,
-            // just turning the motors down is insufficient to land.
-            ctrl->updateMotors(2500);
+            //Were you thinking something like this?  It gets the job done.
+            if(ctrl->getTiltAngle() == 0)
+            {
+                std::cout << "Landing...\n";
+                while(gps->getAltitude() >50.0)
+                    ctrl->updateMotors(2500);
+                while(gps->getAltitude() >25.0)
+                    ctrl->updateMotors(2750);
+                while(gps->getAltitude() >0.0)
+                    ctrl->updateMotors(2900);
+                ctrl->updateMotors(3000);
+            }
```

Comment:
yeah, this works; the kind of downside here is the tight loop that keeps updating the motors, which takes the mutex on the flight controller each time unnecessarily and it will do that hundreds of times per second, so that seems wasteful. Once your RPMs have been set, there's no need to keep resetting them; maybe instead you should set the motors to 2500 BEFORE the loop, and then check in once a second or something by using the sleep command to wait (search the project for `std::this_thread::sleep_for` for an example, the FlightLog and Simulator classes both do something like this).  This would even let you keep printing feedback to the console.  Like you could say on each loop, say " .... [current altitude] ...\n" . to keep the user aware and then sleep for another second. Then when each loop condition is broken readjust the motors up to the next RPM level ONE time and go into another sleep/print loop.  Does that make sense?

Reply:
what about this?